### PR TITLE
[SPARK-10381] Fix mixup of taskAttemptNumber & attemptId in OutputCommitCoordinator (branch-1.3 backport)

### PR DESCRIPTION
**Pull request by JoshRosen**
This is a backport of #8544 to `branch-1.3` for inclusion in 1.3.2.

### CODE REVIEW

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

File: core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ import org.apache.spark.annotation.DeveloperApi
 class TaskInfo(
     val taskId: Long,
     val index: Int,
-    val attempt: Int,
```
Comment (yhuai) [inline core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala]: Maybe we need to take a look at the code to see the meaning of `attempt`.  I find https://github.com/apache/spark/blob/branch-1.3/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L451-L452. Looks like it is indeed `attemptNumber`.

**Comment by yhuai** [comment]
LGTM. We can merge it once jenkins is good.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by yhuai** [comment]
Jenkins, retest this please.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by marmbrus** [comment]
test this please

**Comment by JoshRosen** [comment]
It looks like all of the tests passed on this last run; it just timed out during MiMa checks while fetching some old versions of dependencies. The tests that failed in earlier runs are known to be flaky. Given this, I'm going to merge this now.

**Pull request closed without merging**